src/middlewares/csrf_test.ts

```diff
@@ -0,0 +1,363 @@
+import { App } from "../app.ts";
+import type { FreshContext } from "../context.ts";
+import { csrf } from "./csrf.ts";
+import { expect } from "@std/expect";
+import { spy } from "@std/testing/mock";
+
+const rawSimplePostHandler = async (ctx: FreshContext) => {
+  if (ctx.req.headers.get("content-type") === "application/json") {
+    return new Response((await ctx.req.json())["name"]);
+  } else {
+    return new Response((await ctx.req.formData()).get("name"));
+  }
+};
+
+Deno.test("CSRF by Middleware", async (t) => {
+  let simplePostHandler = spy(rawSimplePostHandler);
```

Comment:
```suggestion
  using simplePostHandler = spy(rawSimplePostHandler);
```
Might not better but maybe best to do.

Reply:
spy doesn’t support the using keyword, so using let seems to be the best solution. Thanks!